### PR TITLE
 Use `sanitize` helper instead of `#html_safe`

### PR DESCRIPTION
**Pull request by no-reply**
Marking unknown strings as `#html_safe` in a view is not safe. This fixes many
instances of this, in favor of using the `sanitize` helper.

Related to #3187; #3229.

Guidance for testing:
* Visit affected pages and ensure formatting remains correct. It's possible that the switch from `html_safe` impacts details of certain markup.
* Especially watch out for any issues with:
  * Batch edit
  * FileSet metadata/characterization views
  * Notifications
  * User activity

@samvera/hyrax-code-reviewers

### CODE REVIEW

File: app/views/hyrax/file_sets/_extra_fields_modal.html.erb [hunk 1]
```diff
@@ -11,7 +11,7 @@
         <h2 id="extraFieldsModal_<%= name %>_Label">Additional <%= label %>(s)</h2>
       </div>
       <div class="modal-body">
-        <%= values.join("<br />").html_safe %>
+        <%= sanitize values.join("<br />") %>
```
[the remark this answers is not in the document]
Comment (no-reply) [reply app/views/hyrax/file_sets/_extra_fields_modal.html.erb]: `safe_join(values, tag(:br))` ?

File: app/views/hyrax/stats/file.html.erb [hunk 2]
```diff
@@ -2,7 +2,7 @@
 <script>
 //<![CDATA[
 
-  var hyrax_item_stats = <%= @stats.to_flot.to_json.html_safe %>;
+  var hyrax_item_stats = <%= sanitize @stats.to_flot.to_json %>;
```
[the remark this answers is not in the document]
Comment (jcoyne) [reply app/views/hyrax/stats/file.html.erb]: That's from our stats table, not from user input.

File: app/views/hyrax/dashboard/collections/_flash_msg.html.erb [hunk 3]
```diff
@@ -2,7 +2,7 @@
   <% if flash[type].present? %>
     <div class="alert <%= flash_dom_class %> alert-dismissable" role="alert">
       <button type="button" class="close" data-dismiss="alert" aria-label="Close"><span aria-hidden="true">&times;</span></button>
-      <%= safe_join(Array.wrap(flash[type]).map(&:html_safe), '<br/>'.html_safe) %>
+      <%= sanitize safe_join(Array.wrap(flash[type]), tag(:br)) %>
```
[the remark this answers is not in the document]
Comment (no-reply) [reply app/views/hyrax/dashboard/collections/_flash_msg.html.erb]: `sanitize Array.wrap(flash[type]).join(tag(:br))` ?

File: app/views/hyrax/batch_edits/edit.html.erb [hunk 4]
```diff
@@ -2,7 +2,7 @@
 
 <div class="scrollx scrolly fileHeight"> <!-- original values -->
   <h3> <b>Changes will be applied to: (<%= @form.names.size %> works) </b></h3>
-   <%= @form.names.join(", ").html_safe %>
+   <%= sanitize @form.names.join(", ") %>
```
[the remark this answers is not in the document]
Comment (no-reply) [reply app/views/hyrax/batch_edits/edit.html.erb]: I think that will escape values in `@form.names`, right? That seems to be something the previous implementation is invested in preventing.

File: app/views/hyrax/stats/file.html.erb (hunk 2 above)
[the remark this answers is not in the document]
Comment (no-reply) [reply app/views/hyrax/stats/file.html.erb]: I'm not really sure at a glance about how confident to be that the stats table isn't susceptible to injection, or that we are writing to it with the idea that the content will eventually be embedded un-sanitized in a <script> tag.

I'm happy to remove these changes for the moment, but I think I'd like to include them in a larger audit of our use of `html_safe` that seems necessary.

File: app/views/hyrax/file_sets/_extra_fields_modal.html.erb (hunk 1 above)
[the remark this answers is not in the document]
Comment (no-reply) [reply app/views/hyrax/file_sets/_extra_fields_modal.html.erb]: Oh, hang on. `safe_join` here will also escape markup from `values`, in a departure from the previous behavior.

Possibly this is fine, but without looking closer at how this is used I'm reluctant to make this (larger) change.

[the remark this answers is not in the document]
Comment (jcoyne) [reply app/views/hyrax/file_sets/_extra_fields_modal.html.erb]: It was added here: https://github.com/samvera/sufia/commit/9d5acd3f2fbf886157bd2841a7d6364c536d5e94 and I'm certain he was just doing this to keep the `<br>` tag from getting escaped.

File: app/views/hyrax/dashboard/collections/_flash_msg.html.erb (hunk 3 above)
[the remark this answers is not in the document]
Comment (jcoyne) [reply app/views/hyrax/dashboard/collections/_flash_msg.html.erb]: 👍

File: app/views/hyrax/stats/file.html.erb (hunk 2 above)
[the remark this answers is not in the document]
Comment (no-reply) [reply app/views/hyrax/stats/file.html.erb]: how about `<%= raw json_escape @stats.to_flot.to_json %>`

File: app/views/hyrax/batch_edits/edit.html.erb (hunk 4 above)
[the remark this answers is not in the document]
Comment (jcoyne) [reply app/views/hyrax/batch_edits/edit.html.erb]: this was introduced here: https://github.com/samvera/sufia/commit/f57234b4ec811cb112070e22c6d5fbf3aa0ef8df#diff-19beaf0eec0782dcad9ac2aba5f6de3aR99 and I'm fairly certain it was unintentional.

**Comment by no-reply** [comment]
@jcoyne i've fixed up some of this in response to your requests.

I ended up doing `raw json_escape` (https://api.rubyonrails.org/classes/ERB/Util.html#method-c-json_escape) for the JSON use case.

I still have questions about your preference for `safe_join` in a couple of places. It seems to escape the non-html_safe strings much more aggressively than the `sanitize` option. I think I have a preference for `sanitize` as more forgiving to applications hoping to provide marked up data.

**Pull request merged**